ui/src/util/configManager.tsx

```diff
@@ -1,17 +1,69 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { InferProps } from 'prop-types';
 
 import { getFormattedMessage } from './messageUtil';
 import { setMetaInfo, setUnifiedConfig } from './util';
 import { loadGlobalConfig } from './script';
 import ErrorModal from '../components/ErrorModal/ErrorModal';
+import { GlobalConfig } from '../types/globalConfig/globalConfig';
 
-class ConfigManager extends Component {
-    constructor(props) {
+interface ConfigManagerState {
+    unifiedConfig: GlobalConfig;
+    appData: IAppData;
+    loading: boolean;
+    syntaxError: boolean;
+    fileNotFoundError?: boolean;
+}
+
+interface IAppData {
+    app: string;
+    custom_rest: string;
+    nullStr: 'NULL';
+    stanzaPrefix: string;
+}
+
+class ConfigManager extends Component<
+    InferProps<typeof ConfigManager.propTypes>,
+    ConfigManagerState
+> {
+    static propTypes = {
+        children: PropTypes.func.isRequired,
+    };
+
+    constructor(props: InferProps<typeof ConfigManager.propTypes>) {
         super(props);
         this.state = {
-            unifiedConfig: {},
-            appData: {},
+            unifiedConfig: {
```

Comment:
maybe it is better to set unifiedConfig optional instead of having ambiguous nullish state of config?